src/types.ts

```diff
@@ -161,17 +161,31 @@ export async function iterativelyCheckStatus(
     txHash: string,
     starknetWrapper: StarknetWrapper,
     resolve: (status: string) => void,
-    reject: (reason: Error) => void
+    reject: (reason: Error) => void,
+    retryCount = 10
 ) {
-    const statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
-        warn(reason);
-        return undefined;
-    });
+    let statusObject;
+    let errorReason;
+    let count = retryCount;
+
+    while (count > 0) {
+        statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
+            errorReason = reason;
+            return undefined;
+        });
+
+        if (statusObject) {
+            break;
+        }
 
-    if (!statusObject) {
         warn("Retrying transaction status check...");
-        // eslint-disable-next-line prefer-rest-params
-        setTimeout(iterativelyCheckStatus, CHECK_STATUS_RECOVER_TIMEOUT, ...arguments);
+        await sleep(CHECK_STATUS_RECOVER_TIMEOUT);
+        count--;
```

Comment:
On the last iteration, we wait for `CHECK_STATUS_RECOVER_TIMEOUT`, only to not retry. The loop could be a `while true`, and in the end we could: decrement, check if count > 0 then either sleep or break out.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/279/commits/57b63a3440197fea0fc2600adf637b11376c2d4f

Reply:
I think there was a misunderstanding, I didn't mean that the countdown loop should cover the whole function body, just that the recursion should be replaced with a loop. And the inner countdown loop should stay, no?